feature/policy_forwarding/otg_tests/mpls_gre_udp_decap_test/mpls_gre_udp_decap_test.go

```diff
@@ -0,0 +1,571 @@
+	default:
+		t.Errorf("Unknown IP type %s in testcase %s", fc.innerIpType, tc.name)
+		return
+	}
+
+	srcIpAddr := net.ParseIP(fc.innerIPSrc)
+	destIpAddr := net.ParseIP(fc.innerIPDest)
+
+	packetCount := 0
+	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
+	for packet := range packetSource.Packets() {
+		if packet.Layer(icmpLayer) != nil {
+			t.Logf("Skipping ICMP packet")
+			continue
+		}
+
+		if packet.Layer(baseIpLayer) == nil {
+			t.Logf("Skipping non %s packet", fc.innerIpType)
+			continue
+		}
+
+		packetCount++
+
+		mplsInnerLayer := packet.Layer(layers.LayerTypeMPLS)
+		mplsPacket, ok := mplsInnerLayer.(*layers.MPLS)
+		if !ok || mplsPacket == nil {
+			t.Errorf("MPLS layer not found for packet %d", packetCount)
+			continue
+		}
+		if mplsPacket.Label != mplsLabel {
+			t.Errorf("MPLS inner packet %d: got label %d, want %d", packetCount, mplsPacket.Label, mplsLabel)
+		}
+
+		switch fc.innerIpType {
+		case IPv4:
+			ipInnerLayer := packet.Layer(layers.LayerTypeIPv4)
+			if ipInnerLayer == nil {
+				t.Errorf("Inner IPv4 layer not found for packet %d", packetCount)
+				continue
+			}
+			ipInnerPacket, ok := ipInnerLayer.(*layers.IPv4)
+			if !ok || ipInnerPacket == nil {
+				t.Errorf("Unable to extract inner IPv4 layer for packet %d", packetCount)
+				continue
+			}
+
+			if !ipInnerPacket.SrcIP.Equal(srcIpAddr) {
+				t.Errorf("IPv4 inner packet %d: got SrcIP %s, want %s", packetCount, ipInnerPacket.SrcIP.String(), srcIpAddr.String())
+			}
+			if !ipInnerPacket.DstIP.Equal(destIpAddr) {
+				t.Errorf("IPv4 inner packet %d: got DstIP %s, want %s", packetCount, ipInnerPacket.DstIP.String(), destIpAddr.String())
+			}
+		case IPv6:
+			ipInnerLayer := packet.Layer(layers.LayerTypeIPv6)
+			if ipInnerLayer == nil {
+				t.Errorf("Inner IPv6 layer not found for packet %d", packetCount)
+				continue
+			}
+			ipInnerPacket, ok := ipInnerLayer.(*layers.IPv6)
+			if !ok || ipInnerPacket == nil {
+				t.Errorf("Unable to extract inner IPv6 layer for packet %d", packetCount)
+				continue
+			}
+			if !ipInnerPacket.SrcIP.Equal(srcIpAddr) {
+				t.Errorf("IPv6 inner packet %d: got SrcIP %s, want %s", packetCount, ipInnerPacket.SrcIP.String(), srcIpAddr.String())
+			}
+			if !ipInnerPacket.DstIP.Equal(destIpAddr) {
+				t.Errorf("IPv6 inner packet %d: got DstIP %s, want %s", packetCount, ipInnerPacket.DstIP.String(), destIpAddr.String())
+			}
+		}
+	}
+	if packetCount != noOfPackets {
+		t.Errorf("Expected %d %s decapsulated packets with inner %s packet, got %d", noOfPackets, tc.encapType, fc.innerIpType, packetCount)
+	} else {
+		t.Logf("Found %d %s decapsulated packets with inner %s packet", packetCount, tc.encapType, fc.innerIpType)
+	}
+}
+
+func configureStaticRoutes(t *testing.T, dut *ondatra.DUTDevice) {
+	for index, destination := range []string{ipv4_inner_dst_A, ipv4_inner_dst_B} {
+		configStaticRoute(t, dut, fmt.Sprintf("%s/32", destination), otgPort2.IPv4, fmt.Sprintf("%d", index))
```

Comment:
For simple use cases, use built-in string append operations and/or `strconv.Itoa()`.  

```suggestion
		configStaticRoute(t, dut, destination+"/32", otgPort2.IPv4, strconv.Itoa(index))
```